tools/app_interface_reporter.py

```diff
@@ -434,6 +434,8 @@ def get_build_history_pool(jenkins_map, jobs,
     for job in result:
         if 'build_history' not in job:
             continue
+        if not job['build_history']:
+            continue
         successes = [h for h in job['build_history'] if h == 'SUCCESS']
```

Comment:
There are three hashtable lookups and two ifs in the happy path here.
A bit of refactoring will do this good. Something like:
```python
build_history = job.get('build_history')
if not build_history:
    continue
successes = [_ for _ in build_history if _ == 'SUCCESS']
```
That reduces the lookups and the ifs to one each.

Reply:
Gotcha! ;D